students/jeremy_m/lesson04_exercises/mailroom_part_2.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+# lesson 03 Exercise - String Formatting Lab
+# Jeremy Monroe
+
+import os
+import pathlib
+import time
+
+donors = {'Charlize Theron': [134000],
+          'Charlie Boorman': [15, 5],
+          'James Franco': [25, 250, 2, 500],
+          'Nike': [22000],
+          'Count Chocula': [1257633, 2532790]
+          }
+
+donor_totals = {}
+
+user_input = ''
+
+messages = {'start': ("What would you like to do?\n"
+                      "1: to Send a Thank You\n"
+                      "2: Create a Report\n"
+                      "3 to create letters for all donors\n"
+                      "quit: to Quit"
+                      ),
+            'thanks': ("Please input the donors name "),
+            'thank_you_message': ("{:^42}\n"
+                                  "{:^42}\n"
+                                  "For your incredibly generous donation of:\n"
+                                  "{:>19}{:<23,}\n\n"),
+            'donor_letter': ("{:^41}\n"
+                             "Thank you so much for your generous donation of:\n"
+                             "{:>21}{:,}\n"
+                             "We will always remember your money fondly."),
+            'letters_confirmation': 'Is this ok?\nType y or n:'
+            }
+
+
+def letters_for_all():
+    """ This function creates a directory in the cwd and fills that with an
+        individual file (letter) for each donor.
+    """
+
+    clear_screen()
+    print('Note: This operation will create a directory in the current working directory')
+    if get_user_input(messages['letters_confirmation']).lower() == 'y':
+        dir_path = os.getcwd() + '/letters'
+        os.mkdir(dir_path)
+        os.chdir(dir_path)
+
+        clear_screen()
+        for donor in donors:
+            donor_filename = '_'.join(donor.split(' ')) + '.txt'
+            with open(donor_filename, 'w+') as new_file:
```

Comment:
you should need the `chdir()` in line 50 if you just include your `dir_path` here as
`with open(f"{dir_path}/{donor_filename}", "w+") as new_file:`

Reply:
Yep... I ran into an issue where the `letters` were being created in the cwd rather than newly created letters directory so I added the `chdir` to fix it. I think I had forgotten the '/' between `dir_path` and `donor_filename` and somehow the files at least wound up in the cwd.

Thanks for the comments on this pull request, helpful corrections.